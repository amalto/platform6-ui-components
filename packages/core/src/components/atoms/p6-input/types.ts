export enum P6InputType {
  email = 0,
  number = 1,
  password = 2,
  search = 3,
  tel = 4,
  text = 5,
  url = 6,
}
