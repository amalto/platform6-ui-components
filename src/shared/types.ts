export type Mode =
  | "danger"
  | "warning"
  | "default"
  | "info"
  | "success"
  | "primary";

export type Size = "small" | "default" | "normal" | "medium" | "large";

export type Position = "top" | "right" | "bottom" | "left";
