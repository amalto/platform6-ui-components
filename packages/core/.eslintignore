dist/
loader/

src/components.d.ts