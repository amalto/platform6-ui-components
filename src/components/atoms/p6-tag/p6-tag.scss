@import "bulma/sass/elements/tag";

:host {
  display: inline-block;
}
