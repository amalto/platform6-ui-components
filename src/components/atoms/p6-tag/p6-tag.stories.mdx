import {
  Story,
  Preview,
  Meta,
  Markdown,
  Description,
} from "@storybook/addon-docs/blocks";

import Readme from "./readme.md";

import { DefaultStory, SizeStory, ModeStory } from "./p6-tag.stories.ts";

<Meta title="Atoms/Tag" component="p6-tag" />

# Tag

<Preview>
  <Story name="Default" height="58px" parameters={DefaultStory.parameters}>
    {DefaultStory()}
  </Story>
</Preview>

<Preview>
  <Story name="Sizes" height="58px" parameters={SizeStory.parameters}>
    {SizeStory()}
  </Story>
</Preview>

<Preview>
  <Story name="Modes" height="58px" parameters={ModeStory.parameters}>
    {ModeStory()}
  </Story>
</Preview>

<Description markdown={Readme} />
