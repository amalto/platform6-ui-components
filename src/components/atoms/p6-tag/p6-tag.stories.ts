import { Components } from "../../../components";
import {
  ComponentProps,
  getElement,
  makeModeStory,
  makeSizeStory,
  makeStory,
  Props,
} from "../../../shared/storybook";
import { Mode, Size } from "../../../shared/types";

const component = "p6-tag";

export default {
  title: "Atoms/Tag",
  component,
};

const componentProps: ComponentProps = ["mode", "size"];

const getStoryField = (
  name: string,
  props?: Props<Components.P6Tag>
): HTMLElement => getElement(component, name, props);

export const Default = makeStory<{
  size: Size;
  mode: Mode;
  label: string;
}>({
  componentProps,
  args: {
    size: Size.normal,
    mode: Mode.default,
    label: "Tag",
  },
  builder: ({ label, ...args }): HTMLElement =>
    getStoryField(label, { ...args }),
});

export const Sizes = makeSizeStory({
  componentProps,
  builder: ({ key, value }) =>
    getStoryField(key, {
      size: value,
    }),
});

export const Modes = makeModeStory({
  componentProps,
  builder: ({ key, value }) =>
    getStoryField(key, {
      mode: value,
    }),
});
