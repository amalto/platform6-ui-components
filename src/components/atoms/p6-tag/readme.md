# p6-tag

<!-- Auto Generated Below -->

## Properties

| Property | Attribute | Description | Type                                                                                       | Default       |
| -------- | --------- | ----------- | ------------------------------------------------------------------------------------------ | ------------- |
| `mode`   | `mode`    | Mode        | `Mode.danger \| Mode.default \| Mode.info \| Mode.primary \| Mode.success \| Mode.warning` | `undefined`   |
| `size`   | `size`    | Size        | `Size.large \| Size.medium \| Size.normal \| Size.small`                                   | `Size.normal` |

---

_Built with [StencilJS](https://stenciljs.com/)_
