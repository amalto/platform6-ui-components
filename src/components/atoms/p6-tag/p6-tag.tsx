import { Component, ComponentInterface, h, JSX, Prop } from "@stencil/core";
import { Mode, Size } from "~shared/types";

@Component({
  tag: "p6-tag",
  styleUrls: ["p6-tag.scss"],
  shadow: true,
})
export class P6Tag implements ComponentInterface {
  /**
   * Mode
   */
  @Prop() mode!: Mode;

  /**
   * Size
   */
  @Prop() size: Size = "small";

  render(): JSX.Element {
    return (
      <span
        class={{
          tag: true,
          [`is-${this.mode}`]: Boolean(this.mode),
          [`is-${this.size}`]: true,
        }}
      >
        <slot />
      </span>
    );
  }
}
