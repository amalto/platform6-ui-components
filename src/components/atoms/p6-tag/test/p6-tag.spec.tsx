import { newSpecPage } from "@stencil/core/testing";
import { Mode, Size } from "~shared/types";
import { P6Tag } from "../p6-tag";

describe("p6-tag", () => {
  it("renders", async () => {
    const page = await newSpecPage({
      components: [P6Tag],
      html: `<p6-tag>Tag</p6-tag>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  it("mode", async () => {
    const page = await newSpecPage({
      components: [P6Tag],
      html: `<p6-tag mode=${Mode.warning}>Warning</p6-tag>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  it("size", async () => {
    const page = await newSpecPage({
      components: [P6Tag],
      html: `<p6-tag size=${Size.large}>Large</p6-tag>`,
    });
    expect(page.root).toMatchSnapshot();
  });
});
