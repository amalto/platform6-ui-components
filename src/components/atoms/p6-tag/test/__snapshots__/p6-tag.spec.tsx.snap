// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`p6-tag mode 1`] = `
<p6-tag mode="1">
  <mock:shadow-root>
    <span class="is-warning tag">
      <slot></slot>
    </span>
  </mock:shadow-root>
  Warning
</p6-tag>
`;

exports[`p6-tag renders 1`] = `
<p6-tag>
  <mock:shadow-root>
    <span class="tag">
      <slot></slot>
    </span>
  </mock:shadow-root>
  Tag
</p6-tag>
`;
